LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionAS923.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaTools.Regions
+{
+    using System;
+    using System.Collections.Generic;
+    using LoRaTools.LoRaPhysical;
+    using LoRaTools.Utils;
+    using LoRaWan;
+
+    public class RegionAS923 : Region
+    {
+        private readonly Hertz frequencyOffset;
+        private readonly bool useDwellTimeLimit;
+
+        public RegionAS923(Hertz frequencyChannel0, Hertz frequencyChannel1, int dwellTime = 0)
+            : base(LoRaRegionType.AS923)
+        {
+            frequencyOffset = new Hertz(frequencyChannel0.AsUInt64 - 923_200_000);
+            if (frequencyChannel1.AsUInt64 - 923_400_000 != frequencyOffset.AsUInt64)
+            {
+                throw new ArgumentException($"Provided channel frequencies {frequencyChannel0}, {frequencyChannel1} for Region {LoRaRegion} are inconsistent.");
+            }
+
+            useDwellTimeLimit = dwellTime == 1;
```

Comment:
if I undertand correctly the value of dwelltime can be either 1 or 0, if the value is above we should throw a ConfigurationErrorsException